### DISCUSSION TITLE
ApplicationContext not work in .net core 3.1

### DISCUSSION THREAD

**Discussion by ratku** [MarimerLLC/csla#2451] [Questions]
Csla.ApplicationContext   not work in .net core 3.1
may be need add  Compile options when  in web aspnetcore project,should use HttpContext  direct  " in code" with IHttpContextAccessor?

**Comment by rockfordlhotka** [comment]
You must use the ApplicationContextManager from the Csla.AspNetCore package to get correct behavior in aspnetcore. The default is to use AsyncLocal, which works in _.NET Core_, but not in _aspnetcore_.
